tests/test_update.py

```diff
@@ -27,28 +27,27 @@ async def test_software_update_properties(hass: HomeAssistant) -> None:
     version = car_mock_data.VEHICLE_DATA["vehicle_state"]["software_update"]["version"]
 
     state = hass.states.get("update.my_model_s_software_update")
-    # assert state.state == str(car_mock_data.VEHICLE_STATE["software_update"])
-
-    assert state.attributes.get("latest_version") == version
+    assert state.state == "on"
+    assert state.attributes.get("latest_version") == version + " (Available to install)"
```

Comment:
Do you mind adding a test confirming #356 is fixed? I know your fix addressed it just by looking at the code, but I like to get tests in if they were previously reported issues so we don't get a regression in the future.

Reply:
Can do, I was hoping to iterate through each status in a for loop to test everything, but I'm failing to get the state to update mid-test after updating the mock data. Not sure if something to do with the coordinator or what.

something like this, but even re-running setup_platform doesn't update the values after the first time

```python
    for raw, pretty in status_strings.items():
        car_mock_data.VEHICLE_DATA["vehicle_state"]["software_update"]["status"] = raw
        # Something here to update states?
        state = hass.states.get("update.my_model_s_software_update")
        assert state.attributes.get("latest_version") == f"{version} ({pretty})"
        # assert in-progress, installable, etc...
```

If you have any tips I'm all ears. I'll keep digging though.

Reply:
Ok the issue is you're just updating the mock data which is basically the API. You haven't ingested it into the component yet unless you call an update on the component to reach out to the `API`. Other tests are doing it by just setting up the component again which forces a `pull` from the API using the new mocked data.

You could try looking at HA tests to see how people are doing updates. That said, I can live with the idea of you changing the mock data, and then having the entity setup and then test the value. That at least covers all the interesting startup states.

Reply:
> Ok the issue is you're just updating the mock data which is basically the API. You haven't ingested it into the component yet unless you call an update on the component to reach out to the `API`. Other tests are doing it by just setting up the component again which forces a `pull` from the API using the new mocked data.
> 
> You could try looking at HA tests to see how people are doing updates. That said, I can live with the idea of you changing the mock data, and then having the entity setup and then test the value. That at least covers all the interesting startup states.

After looking at several example tests in the HA repo, it seems what I was trying to do isn't very typical. So I went the route to setup for each state.
Unfortunately doing a setup twice in the same test function didn't overwrite the previous state either, so I created a test function for each phase of the update status. It's probably a little overkill now tbh.

Reply:
Ok thanks.